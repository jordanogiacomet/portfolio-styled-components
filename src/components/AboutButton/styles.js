import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
