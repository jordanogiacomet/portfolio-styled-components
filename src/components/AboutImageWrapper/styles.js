import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
