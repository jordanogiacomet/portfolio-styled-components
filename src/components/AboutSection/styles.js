import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
