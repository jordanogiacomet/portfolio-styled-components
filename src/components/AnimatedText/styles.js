import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;