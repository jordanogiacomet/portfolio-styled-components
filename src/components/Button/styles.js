import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 