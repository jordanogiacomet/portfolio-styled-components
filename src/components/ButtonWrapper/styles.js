import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;