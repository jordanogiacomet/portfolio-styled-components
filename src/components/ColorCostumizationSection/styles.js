// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
