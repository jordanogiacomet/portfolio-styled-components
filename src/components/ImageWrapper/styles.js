import styled, { css } from 'styled-components';

const aboutImageWrapperStyles = css`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;

const heroImageWrapperStyles = css`
  border-radius: 9999px;
  background-color: #181818;
  width: 250px;
  height: 250px;
  position: relative;

  @media ${({ theme }) => theme.media.lg} {
    width: 400px;
    height: 400px;
  }

  img {
    position: absolute;
    transform: translate(-50%, -50%);
    top: 50%;
    left: 50%;
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: 9999px;
  }
`;

const getSectionStyles = (type) => {
  switch (type) {
    case 'about-wrapper':
      return aboutImageWrapperStyles
    case 'hero-wrapper':
      return heroImageWrapperStyles 
    default:
      break;
  }
}

export const Container = styled.div`
  ${({ type }) => css`
    ${getSectionStyles(type)}
  `}
`;