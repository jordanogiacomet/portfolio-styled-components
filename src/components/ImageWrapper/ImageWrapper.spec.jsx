import { renderTheme } from '../../styles/render-theme';
import { ImageWrapper } from '.';
import { screen } from '@testing-library/react';
import 'jest-styled-components';

describe('<ImageWrapper />', () => {
  it('should render ImageWrapper with children', () => {
    renderTheme(
      <ImageWrapper type="hero-wrapper">
        <img src="/images/hero-image.webp" alt="hero image" />
      </ImageWrapper>
    );

    expect(screen.getByAltText('hero image')).toBeInTheDocument();
  });
 /*
  it('should apply the correct styles for type "hero-wrapper"', () => {
    const { container } = renderTheme(
      <ImageWrapper type="hero-wrapper">
        <img src="/images/hero-image.png" alt="hero image" />
      </ImageWrapper>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('border-radius', '9999px');
    expect(wrapper).toHaveStyleRule('background-color', '#181818');
    expect(wrapper).toHaveStyleRule('width', '250px');
    expect(wrapper).toHaveStyleRule('height', '250px');

    const img = screen.getByAltText('hero image');
    expect(img).toHaveStyleRule('position', 'absolute');
    expect(img).toHaveStyleRule('transform', 'translate(-50%, -50%)');
    expect(img).toHaveStyleRule('top', '50%');
    expect(img).toHaveStyleRule('left', '50%');
    expect(img).toHaveStyleRule('width', '100%');
    expect(img).toHaveStyleRule('height', '100%');
    expect(img).toHaveStyleRule('object-fit', 'cover');
    expect(img).toHaveStyleRule('border-radius', '9999px');
  });

  it('should apply the correct styles for type "about-wrapper"', () => {
    const { container } = renderTheme(
      <ImageWrapper type="about-wrapper">
        <img src="/images/about-image.png" alt="about image" />
      </ImageWrapper>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('margin-top', '4px');
    expect(wrapper).toHaveStyleRule('order', '2');

    const img = screen.getByAltText('about image');
    expect(img).toHaveStyleRule('display', 'block');
    expect(img).toHaveStyleRule('max-width', '100%');
    expect(img).toHaveStyleRule('height', 'auto');
    expect(img).toHaveStyleRule('border-radius', '10px');
    expect(img).toHaveStyleRule('box-shadow', '0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1)');
  });

 */ 
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ImageWrapper type="hero-wrapper">
        <img src="/images/hero-image.webp" alt="hero image" />
      </ImageWrapper>
    );

    expect(container).toMatchSnapshot();
  });
});
