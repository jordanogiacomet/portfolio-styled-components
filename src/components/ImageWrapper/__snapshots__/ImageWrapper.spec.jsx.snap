// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageWrapper /> should match snapshot 1`] = `
.c0 {
  border-radius: 9999px;
  background-color: #181818;
  width: 250px;
  height: 250px;
  position: relative;
}

.c0 img {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 9999px;
}

@media (min-width: 1024px) {
  .c0 {
    width: 400px;
    height: 400px;
  }
}

<div>
  <div
    class="c0"
    type="hero-wrapper"
  >
    <img
      alt="hero image"
      src="/images/hero-image.png"
    />
  </div>
</div>
`;
